File: RxUI/src/main/java/com/vondear/rxui/animation/RxShineAnimator.java
package com.vondear.rxui.animation;

import android.animation.ValueAnimator;
import android.graphics.Canvas;

import com.vondear.rxui.view.likeview.tools.RxShineView;
import com.vondear.rxui.view.likeview.tools.ei.RxEase;
import com.vondear.rxui.view.likeview.tools.ei.RxEasingInterpolator;

/**
 * @author vondear
 * @date 2016/7/5 下午5:09
 */
public class RxShineAnimator extends ValueAnimator {

    float MAX_VALUE = 1.5f;
    long ANIM_DURATION = 1500;
    Canvas canvas;

    public RxShineAnimator() {
        setFloatValues(1f, MAX_VALUE);
        setDuration(ANIM_DURATION);
        setStartDelay(200);
        setInterpolator(new RxEasingInterpolator(RxEase.QUART_OUT));
    }

    public RxShineAnimator(long duration, float maxValue, long delay) {
        setFloatValues(1f, maxValue);
        setDuration(duration);
        setStartDelay(delay);
        setInterpolator(new RxEasingInterpolator(RxEase.QUART_OUT));
    }

    public void startAnim(final RxShineView rxShineView, final int centerAnimX, final int centerAnimY) {

        start();
    }

    public void setCanvas(Canvas canvas) {
        this.canvas = canvas;
    }


}

File: RxUI/src/main/java/com/tamsiree/rxui/animation/RxShineAnimator.kt
package com.tamsiree.rxui.animation

import android.animation.ValueAnimator
import android.graphics.Canvas
import com.tamsiree.rxui.view.likeview.tools.RxShineView
import com.tamsiree.rxui.view.likeview.tools.ei.RxEase
import com.tamsiree.rxui.view.likeview.tools.ei.RxEasingInterpolator

/**
 * @author tamsiree
 * @date 2016/7/5 下午5:09
 */
class RxShineAnimator : ValueAnimator {
    var MAX_VALUE = 1.5f
    var ANIM_DURATION: Long = 1500
    var canvas: Canvas? = null

    constructor() {
        setFloatValues(1f, MAX_VALUE)
        duration = ANIM_DURATION
        startDelay = 200
        interpolator = RxEasingInterpolator(RxEase.QUART_OUT)
    }

    constructor(duration: Long, maxValue: Float, delay: Long) {
        setFloatValues(1f, maxValue)
        setDuration(duration)
        startDelay = delay
        interpolator = RxEasingInterpolator(RxEase.QUART_OUT)
    }

    fun startAnim(rxShineView: RxShineView?, centerAnimX: Int, centerAnimY: Int) {
        start()
    }

}